src/network/bpf/sysctl_monitor/sysctl-write-event.h

```diff
@@ -0,0 +1,14 @@
+/* SPDX-License-Identifier: LGPL-2.1-or-later */
+
+#pragma once
+
+#ifndef TASK_COMM_LEN
+#define TASK_COMM_LEN 16
+#endif
+
+struct sysctl_write_event {
+        char comm[TASK_COMM_LEN];
+        char name[64];
+        char current[32];
+        char newvalue[32];
```

Comment:
these sizes, how do they come to be? 

the comm size is fine, but the others? aren't they are a bit short?

Reply:
taking IF_NAMESIZ as 16 into account the longest valid sysctl path for the network hierarchy i could come up with is:

net/ipv6/conf/123456789012345/mldv1_unsolicited_report_interval

that's a realy sysctl attribute. hence sizing the field to 64 seems too short?

also, in future there might be longer ones, who knows. maybe at least define a clear behaviour if we hit longer names or values: i.e. maybe let's say that if the strings take up the whole array length without being NUL terminated this should indicate they were truncated. And if they are NUL terminated, then userspace knows for sure they are complete? And document that in a comment here?

Updating bpf programs during runtime is not easy, given that the maps they come with might already be populated. Maybe do some trivial preparation here for a future where we might need to update userspace without updating the bpf programs: maybe add a version byte as first field (`uint8_t version`) and set that to some value.

we haven#t done this with the other bpf programs so far, and it is biting us in the ass, since updating systemd at runtime without a strategy on the bpf programs is nasty.

hence maybe let's do better here.

Reply:
the longest sysctl value i could find is in /proc/sys/core/netdev_rss_key btw, with 156 bytes. hence, here too, question is if 32 is really the right choice for the this. maybe also go for 256 on that?

Reply:
I already did this calculations, but the problem is that the stack in eBPF programs is very limited, so I trimmed the size to a reasonable value.

With name, current and newvalue set to 128, 256, 256 I get this: 
```
   62 |         struct vals vals = { };
      |                     ^
error: Looks like the BPF stack limit is exceeded.
Please move large on stack variables into BPF per-cpu array map.
For non-kernel uses, the stack can be increased using -mllvm -bpf-stack-size.
```

Reply:
hmm, no idea what that error means, but i guess it is fundamental to document these things in a comment, since the reasons exist but are not obvious.

I guess truncating the value is less of an issue than truncating the argument names, hence maybe grow the name field to make sure every possible sysctl path we are aware of now will fit?

Reply:
These sizes fits in the stack:
```
struct vals {
        char comm[TASK_COMM_LEN];
        char name[80];
        char current[160];
        char newvalue[160];
};
```
Alternatively, I propose to drop `current`, as we already have the shadow with the value set by us

Reply:
the suggested structure sounds fine to me.

but what i actually care about more is that you add a comment explaining how these values came to be. the precise values should make rough sense, but i don't care too much if one gets 10 bytes more or 10 bytes less. it's the comments explaining thing i care about.

Reply:
oh, and please add a one-byte version id, so that compat between loaded/pinned bpf prog and newer userspace can be kept.

Reply:
Got it.
From the userspace part instead, should I check that the version is 1, so to avoid running an older networkd wit a newer bpf file by mistake?
I think that this is enough:
```c
if (we->version != 1)
        return log_warning_errno(SYNTHETIC_ERRNO(EINVAL),
                        "Invalid sysctl write event version: %d", we->version);
```

Reply:
yeah. something like that.